### ISSUE TITLE
Security headers not enabled by default when using config.toml

### ISSUE THREAD

**Issue by mac-chaffee** [joseluisq/static-web-server#210]
**Describe the bug**
When using a config file instead of CLI flags like this:

```
[general]
http2 = true
http2-tls-cert = "local.dev_cert.pem"
http2-tls-key= "local.dev_key.pem"
log-level = "trace"
```
Security headers don't get enabled:

```
2023-05-29T17:21:52.159831Z  INFO static_web_server::server: security headers: enabled=false
```

**Expected behavior**
Security headers should be enabled automatically when using http2 per the docs: https://static-web-server.net/features/security-headers/

**Environment and Specs**
 - static-web-server: 2.16.0 (built from commit 0f66443)
 - OS: Linux

**Additional context**
I'm suspecting some relation to #209. Something buried in the clap/structopt code is mishandling booleans I think, especially in `default_value_if` which has changed in clap v4 to take different type parameters: https://docs.rs/clap/4.3.0/clap/struct.Arg.html#method.default_value_if

**Comment by joseluisq** [comment]
> I'm suspecting some relation to #209. Something buried in the clap/structopt code is mishandling booleans I think, especially in `default_value_if` which has changed in clap v4 to take different type parameters: https://docs.rs/clap/4.3.0/clap/struct.Arg.html#method.default_value_if

For now, the workaround for this should be just enabling the `security-headers` explicitly in the TOML file along with `http2`.

```toml
[general]
# ....

http2 = true
http2-tls-cert = "local.dev_cert.pem"
http2-tls-key= "local.dev_key.pem"
log-level = "trace"

# Opt-in security headers explicitly
security-headers = true
```

BTW I will update the docs to mention this until we find a solution.

**Comment by joseluisq** [comment]
`security-headers` should be enabled as expected.

**Comment by joseluisq** [comment]
Fixed on [v2.18.0](https://github.com/static-web-server/static-web-server/releases/tag/v2.18.0)
